examples/plot_plotting_metrics_with_error.py

```diff
@@ -0,0 +1,209 @@
+
+"""
+================================
+Plotting Metrics with Error Bars
+================================
+"""
+# %%
+# Load and preprocess the data set
+# ================================
+# We download the data set using `fetch_openml` function in `sklearn.datasets`.
+# The original Adult data set can be found at https://archive.ics.uci.edu/ml/datasets/Adult
+# There are some caveats to using this dataset, but we will use it solely as an example
+# to demonstrate the plotting metrics with error bars functionality
+#
+# We start by importing the various modules we're going to use:
+
+# TODO: Remove in final version (helps run with local packages)
+import matplotlib.pyplot as plt
+from fairlearn.experimental.enable_metric_frame_plotting import plot_metric_frame
+from fairlearn.metrics import MetricFrame
+from sklearn.datasets import fetch_openml
+from sklearn.metrics import recall_score, accuracy_score, confusion_matrix
+from sklearn.tree import DecisionTreeClassifier
+from sklearn.model_selection import train_test_split
+import pandas as pd
+import numpy as np
+import os
+import sys
+nb_dir = os.path.split(os.getcwd())[0]
+if nb_dir not in sys.path:
+    sys.path.append(nb_dir)
+
+
+data = fetch_openml(data_id=1590, as_frame=True)
+X = data.data
+y = (data.target == '>50K') * 1
+X_train, X_test, y_train_true, y_test_true = train_test_split(X, y, test_size=0.33, random_state=42)
+
+X_train_processed = pd.get_dummies(X_train)
+X_test_processed = pd.get_dummies(X_test)
+test_set_sex = X_test['sex']
+
+# %%
+# Error Metrics
+# =============
+# We have many different choices for error metrics. In this notebook we'll just be using a
+# `Normal approximation interval <https://en.wikipedia.org/wiki/Binomial_proportion_confidence_interval#Normal_approximation_interval>` (symmetric)
+# and a `Wilson score interval <https://en.wikipedia.org/wiki/Binomial_proportion_confidence_interval#Wilson_score_interval>` (asymmetric).
+
+# We aim to create a 95% confidence interval, so we use a `z_score` of 1.959964
+z_score = 1.959964
+digits_of_precision = 4
+
+
+def wilson(p, n, digits=digits_of_precision, z=z_score):
+    """Return lower and upper bound"""
+    denominator = 1 + z**2/n
+    centre_adjusted_probability = p + z*z / (2*n)
+    adjusted_standard_deviation = np.sqrt((p*(1 - p) + z*z / (4*n)))/np.sqrt(n)
+    lower_bound = (centre_adjusted_probability - z*adjusted_standard_deviation) / denominator
+    upper_bound = (centre_adjusted_probability + z*adjusted_standard_deviation) / denominator
+    return (round(lower_bound, digits), round(upper_bound, digits))
+
+
+def compute_error_metric(metric_value, sample_size, z_score):
+    """Compute Standard Error Calculation (for Binary Classification)
+
+    Assumes infinitely large population,
+    Should be used when the sampling fraction is small.
+    For sampling fraction > 5%, may want to use finite population correction
+    https://en.wikipedia.org/wiki/Margin_of_error
```

Comment:
Is this a reference for reading up more about it, or a link that backs up the description above? In any case, adding a section as follows would be nice:
```
References
    ----------
    .. [1] https://en.wikipedia.org/wiki/Margin_of_error
```
and also reference it in your description so that it's clear how it relates to the description as `[1]_`